File: backend/src/main/java/com/campus/Campus_Connect/features/honor/dto/response/HonorResponse.java
package com.campus.Campus_Connect.features.honor.dto.response;

import com.campus.Campus_Connect.features.honor.enums.HonorType;
import lombok.Builder;
import lombok.Getter;

import java.time.Instant;

@Getter
@Builder
public class HonorResponse {

    private Integer honorId;

    private HonorType type;

    private String title;

    private String subtitle;

    private String iconUrl;

    private Integer eventId;

    private Integer priority;

    private Instant awardedAt;
}

File: android-app/app/src/main/java/com/example/campusconnect/feature/profile/data/remote/response/HonorResponse.kt
package com.example.campusconnect.feature.profile.data.remote.response

import com.google.gson.annotations.SerializedName

data class HonorResponse(

    // Honor Item -------------------------------------------------------

    @SerializedName("honor_id")
    val honorId: String,

    @SerializedName("type")
    val type: String,                 // badge | medal

    @SerializedName("title")
    val title: String,

    @SerializedName("subtitle")
    val subtitle: String?,

    @SerializedName("icon_url")
    val iconUrl: String?,

    @SerializedName("condition")
    val condition: String?,

    @SerializedName("event_id")
    val eventId: String?,

    // User Honor -------------------------------------------------------

    @SerializedName("priority")
    val priority: Int,

    @SerializedName("awarded_at")
    val awardedAt: String,
)